AppliancesModel/Models/AppliancesDistribution.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AppliancesModel.Contracts;
+using AppliancesModel.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AppliancesModel
+{
+    public class AppliancesDistribution : IAppliancesDistribution
+    {
+        private readonly IAppliances stockContext;
+
+        public AppliancesDistribution(IAppliances stock)
+        {
+            stockContext = stock;
+        }
+
+        public int RefreshStock(Appliance goods, int count)
+        {
+            if (goods.Amount == count)
+            {
+                stockContext.Stock.Remove(goods);
+                return goods.Amount;
+            }
+            else
+            {
+                goods.Amount -= count;
+                return count;
+            }
+        }
+
+        public Appliance CheckGoodsExistance(string applianceName)
+        {
+            foreach (Appliance goods in stockContext.Stock)
```

Comment:
goods -> good
it is one good on each loop

Reply:
or you can use linq
something like this
return stockContext.Stock.FirstOrDefault(x => x.Name == applianceName);